apisix/plugins/workflow.lua

```diff
@@ -157,5 +158,22 @@ function _M.access(conf, ctx)
     end
 end
 
+function _M.log(conf, ctx)
+    for _, rule in ipairs(conf.rules) do
+        local match_result = true
+        if rule.case then
+            local expr, _ = expr.new(rule.case)
```

Comment:
performance: we'd better to cache the `expr` and reuse it

Reply:
same issue in `access` phase:

https://github.com/apache/apisix/pull/12465/files#diff-fdd97cfadd9b485076ddd89c4611747faf1f6c4cb6e833c07729a9e1aa70844eL149